### ISSUE TITLE
IE9 - No Living Connections

### ISSUE THREAD

**Issue by incorvia** [elastic/elasticsearch-js#96]
I'm using the angular.js module integration for the ElasticSearch library in my companies website.  The application works fine on > IE10  but when I load the application in IE9, I get the dreaded 'No Living Connections' error.  

In the documentation for ElasticSearch it states: "We also test in IE 9, but there is a known issue with making non-standard (get/post) requests across domains.".  This may be due to the fact that 'standard' requests are the only requests allowed by XDomainRequest which, I believe IE9 uses for cross domain requests:  

http://blogs.msdn.com/b/ieinternals/archive/2010/05/13/xdomainrequest-restrictions-limitations-and-workarounds.aspx

There was some thought that in order to establish the connection or keep it alive, a 'HEAD' request would need to be made and not being able to make a XDomainRequest via HEAD could be the issue.  The thought was that perhaps forcing the client.search method to use a 'GET' request by setting the 'method' parameter would in fact fix it.

In testing it seems that instantiating a new client and making the most generic match_all query with a 'GET' request fails

```
$: client = elasticsearch.Client({ host: ELASTICSEARCH_URL})
=> EsApiClient {close: function, transport: Transport, cluster: ClusterNS, indices: IndicesNS, _namespaces: Array[2]…}
$: client.search({method: 'GET', body: JSON.stringify({query: {match_all: {}}})})
=> undefined
=> Uncaught TypeError: Cannot read property 'reject' of undefined elasticsearch.js?body=1:27265
```

Note that using a 'POST' will correctly return a promise and I've redacted my elasticsearch_url but did use an appropriate one.  I know that there is a limitation in Javascript that prevents adding data to the body of a 'GET' request and depending on how this library internally handles the 'body' parameter this may be an issue it is running up against.

In short, if we were able to use a 'GET' request (which I'm not convinced we are able to) I'm not sure that that will fix the initial issue of 'No Living Connections'.  I know that it is stated that IE9 is supported but I do not see how it can be given the limitations I am facing.  If it is not capable of being supported, then I would suggest it should be mentioned in the docs.

Any help is appreciated!

**Comment by obashmat** [comment]
Hi guys,
Does anyone have any update on this topic? 
Also currently  <a href="https://gist.github.com/spenceralger/2fa226f91582b0bbac3c">this gist</a> with the compatibility connector does not exist :( 

Thank you for any update.

**Comment by nicoletaCh** [comment]
Hi guys,

I am also intrested in finding the compatibility connector. Could anyone help?

Thanks!!

**Comment by timraymond** [comment]
@obashmat @nicoletaCh I've made it available here: https://gist.github.com/timraymond/042cbed0ef0a9fe3ef30
